### PR TITLE
Add `params_to_tune` for `DeepARModel` and `TFTModel`

### LINKED ISSUE #1197: Add params_to_tune for pf NNs (`DeepARModel`, `TFTModel`)
### 🚀 Feature Request

Add params_to_tune for pf NNs (`DeepARModel`, `TFTModel`)

### Proposal

Suggested grid for `DeepARModel`:
```
{
    "hidden_size": IntDistribution(4, 64, step=4),
    "rnn_layers": IntDistribution(4, 3, step=1),
    "dropout": Uniform(0, 0.5),
    "lr": LogUniformDistribution(1e-5, 1e-2),
}
```

Suggested grid for `TFTModel`:
```
{
    "hidden_size": IntDistribution(4, 64, step=4),
    "lstm_layers": IntDistribution(4, 3, step=1),
    "dropout": Uniform(0, 0.5),
    "attention_head_size": IntDistribution(2, 8, step=2),
    "lr": LogUniformDistribution(1e-5, 1e-2),
}
```

To discuss:
- Is tuning `lr` necessary?

### Test cases

Look at https://github.com/tinkoff-ai/etna/issues/1184.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Mr-Geekman**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Proposed Changes
Look at #1197.

## Closing issues
Closes #1197.

### CODE REVIEW

File: tests/test_models/nn/test_deepar.py [hunk 1]
```diff
@@ -191,3 +192,15 @@ def test_repr():
 def test_deepar_forecast_throw_error_on_return_components():
     with pytest.raises(NotImplementedError, match="This mode isn't currently implemented!"):
         DeepARModel.forecast(self=Mock(), ts=Mock(), prediction_size=Mock(), return_components=True)
+
+
+def test_params_to_tune():
+    model = DeepARModel(decoder_length=3, encoder_length=4)
+    grid = model.params_to_tune()
+    # we need sampler to get a value from distribution
+    sampler = RandomSampler()
```
Comment (brsnw250) [inline tests/test_models/nn/test_deepar.py:201]: Use `seed` for reproducibility

File: tests/test_models/nn/test_tft.py [hunk 2]
```diff
@@ -196,3 +197,15 @@ def test_repr():
 def test_tft_forecast_throw_error_on_return_components():
     with pytest.raises(NotImplementedError, match="This mode isn't currently implemented!"):
         TFTModel.forecast(self=Mock(), ts=Mock(), prediction_size=Mock(), return_components=True)
+
+
+def test_params_to_tune():
+    model = TFTModel(decoder_length=3, encoder_length=4)
+    grid = model.params_to_tune()
+    # we need sampler to get a value from distribution
+    sampler = RandomSampler()
```
Comment (brsnw250) [inline tests/test_models/nn/test_tft.py:206]: same about `seed`

**Review by brsnw250 — CHANGES_REQUESTED** [review]

File: tests/test_models/nn/test_deepar.py (hunk 1 above)
Comment (Mr-Geekman) [reply tests/test_models/nn/test_deepar.py:201]: Will be added in #1209.

File: tests/test_models/nn/test_tft.py (hunk 2 above)
Comment (Mr-Geekman) [reply tests/test_models/nn/test_tft.py:206]: Will be added in #1209.

**Review by brsnw250 — APPROVED** [review]

**Pull request merged**